### PR TITLE
Add feature gate in global config

### PR DESCRIPTION
**Pull request by hzxuzhonghu**

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
/assign @rshriram 

Added `LOCALITY_AWARE_ROUTING ` in initial pr, more features should be moved here in future.

File: mesh/v1alpha1/config.proto [hunk 1]
```diff
@@ -419,3 +422,19 @@ message LocalityLoadBalancerSetting{
...
+  LOCALITY_AWARE_ROUTING = 0;
+}
+
+// Feature specify to enable or disable a specific feature.
+message Feature {
+  // Feature is the name of the feature.
+  FeatureName feature = 1;
+
+  // Specify true to enable the feature or false to disable the feature.
+  bool enable = 2;
```
Comment (rshriram) [inline mesh/v1alpha1/config.proto]: sorry for delay. This is very nice. But do you think its useful to have a enable/disable? 
Why can't we simply have a repeated enum such that in mesh config we have

```
features:
- LOCALITY_AWARE_ROUTING
- FOO_BAR
- BAZ
...
```

If someone wants to disable it, they could remove that line right?

Comment (hzxuzhonghu) [reply mesh/v1alpha1/config.proto]: Originally, i have thought about this, but this requires user to explicitly set all features if they want to enable them. 

How about we make this as type map[string]bool. If we have default enabled features in pilot like 
LOCALITY_AWARE_ROUTING,  FOO_BAR, BAZ, and we want to disable BAZ, we just need to set:

```
features:
  BAZ: false
```

Comment (hzxuzhonghu) [reply mesh/v1alpha1/config.proto]: And also have `Feature1` disabled by default in pilot code, and we want to enable it, then just need explicitly set:

```
features:
  Feature1: true
```

Comment (hzxuzhonghu) [reply mesh/v1alpha1/config.proto]: But this has a limit, the key of map in proto can not be a enum type.

Comment (rshriram) [reply mesh/v1alpha1/config.proto]: hmm.. I think then what you have is good enough.

**Pull request merged**